static/app/components/events/interfaces/spans/spanTreeModel.tsx

```diff
@@ -320,12 +337,6 @@ class SpanTreeModel {
       }
     );
 
-    for (const hiddenSpanGroup of hiddenSpanGroups) {
```

Comment:
I moved this edge case to be before we reduce on `descendantsSource`, since we can just return an empty array without having to visit all the spans in `descendantsSource`.

This is an optimization refactor.